app/src/main/java/com/tristanwiley/chatse/chat/ChatActivity.kt

```diff
@@ -458,13 +459,13 @@ class ChatActivity : AppCompatActivity(), ServiceConnection, RoomAdapter.OnItemC
 
     override fun onItemClick(chatRoom: ChatRoom) {
 
-        val roomNo = chatRoom.num
-        if ((roomNum!=-1) && roomNum != roomNo) {
-            roomNum = roomNo
+        val clickedRoomNo = chatRoom.num
+        if ((currentroomNum != null) && currentroomNum != clickedRoomNo) {
+            currentroomNum = currentroomNum
             doAsync {
                 addChatFragment(createChatFragment(chatRoom))
             }
-            prefs.edit().putString("lastRoomSite", chatRoom.site).putInt("lastRoomNum",roomNo).apply()
+            prefs.edit().putString("lastRoomSite", chatRoom.site).putInt("lastRoomNum",clickedRoomNo).apply()
```

Comment:
Use the constants here as well. Currently here we have lastRoomSite and the constant is last_room_site